netket/experimental/geometry/continuous_geometry.py

```diff
@@ -0,0 +1,122 @@
+        the standard basis is assumed, e.g. (1,0), (0,1) for 2D space.
+
+        Args:
+            dim: (Optional) The number of spatial dimensions of the physical space. If None, a basis has to be
+                            specified. If int and basis is None, the standard basis is assumed.
+            basis: (Optional) A basis for the physical space. If basis is None, the standard basis is assumed.
+        """
+        if dim is None and basis is None:
+            raise ValueError(
+                """Specify either the dimension of the geometry or provide a basis for it."""
+            )
+
+        if basis is None:
+            basis = jnp.eye(dim)
+
+        self._basis = basis
+        self._dim = self._basis.shape[0]
+
+        self._hash = None
+
+    @property
+    def dim(self):
+        r"""The number of spatial dimensions of the geometry."""
+        return self._dim
+
+    @property
+    def basis(self):
+        r"""The basis spanning the geometry."""
+        return self._basis
+
+    @property
+    def inv_basis(self):
+        r"""The reciprocal space of the given geometry."""
+        return jnp.linalg.inv(self.basis)
+
+    @abc.abstractmethod
+    def distance(self, x, y=None, norm=False, mode=None, tri=False) -> list[list]:
+        r"""Distance function for the given geometry. For free space this is just the Euclidean distance. For
+        periodic geometries, this could be Minimum image convention or smooth periodic distance.
+           Args:
+               x: Array (..., N, dim), batch of samples containing N positions with the same dimension as the geometry.
+               y: (Optional) Array (..., M, dim), batch of samples containing M positions with the same dimension
+                                as the geometry. y = None means distance(x,x) with a safe norm.
+               norm: bool, indicates whether to return only distance vectors or also their norm.
+               mode: str, indicates which distance is computed. Options: 'Euclidean' is the standard distance.
+                            'MIC' denotes the minimum image distance. 'Periodic' denotes a smooth periodic distance.
+               tri: bool, indicates whether the whole distance matrix is returned or its upper triangular part.
+           Returns: Array (..., N, M, dim), Distance matrix where the ij-th entry contains the distance between
+                                            the particle in position x[...,i,:] and in position y[...,j,:].
+        """
+
+    @abc.abstractmethod
+    def add(self, x, y):
+        r"""Given two batches of position vectors in the geometry, this function defines how these vectors are added
+        together. For free space, this is the usual vector addition. FOr periodic space, this includes folding particle
+        positions back to the geometry defined."""
+
+    @abc.abstractmethod
+    def random_init(self, shape):
```

Comment:
I wouldn't call this random_init, rather just random_uniform or similar? (the init part is not relevant from the Geometry point of view)

Reply:
For free space I just return zeros, so it is also not really uniform (also not random actually).. So both names are a bit misleading maybe